Add decimation base as input to cog_translate

mccarthyryanc opened the pull request:
Similar to the `decimation_base` added to [morecantile](https://github.com/developmentseed/morecantile/pull/146). I'd like the ability to define a custom `decimation_base` when creating COG overviews. This PR adds that var in along with some tests. For now, it requires an input for `overview_level`, since that defaults to the value returned by `rasterio.rio.overview.get_maximum_overview_level` which uses a [hardcoded base of 2](https://github.com/rasterio/rasterio/blob/842743055124b1b4ef0f43907467a3a95b8fdb82/rasterio/rio/overview.py#L55).

vincentsarago commented:
@mccarthyryanc thanks for the PR 🙏 

Would you be able to update the documentation for this new option 🙏

mccarthyryanc commented:
@vincentsarago , I've updated the docs. Since the docs mostly refer to the CLI (at least in the Advanced Topics section) I also added this option, `rio cogeo --decimation-base`. Hopefully that is okay.

vincentsarago commented on docs/docs/Advanced.md:
````diff
@@ -41,6 +41,20 @@ print(overviews)
+```
+
+This is primarily useful when working with [custom TileMatrixSets](https://developmentseed.org/morecantile/usage/#define-custom-grid) that also use a non-default decimation base.
+
````
😍
